pkg/argo/fetch.go

```diff
@@ -45,38 +43,57 @@ func newFetchResponse(newReq *pb.FetchRequest, eof bool, logs []string) *pb.Fetc
 // Fetch fetches the workflow log and status,
 // design doc: https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/argo_workflow_on_sqlflow.md
 func Fetch(req *pb.FetchRequest) (*pb.FetchResponse, error) {
-	// TODO(yancey1989): fetching running workflow logs
-	wf, err := waitUntilComplete(req)
+	wf, err := k8sReadWorkflow(req.Job.Id)
 	if err != nil {
 		return nil, err
 	}
 
-	stepGroupName, err := getCurrentStepGroup(wf, req.Job.Id, req.StepId)
+	if isWorkflowPending(wf) {
```

Comment:
Is there a way to add test to cover wokflow/pod pending branches?

Reply:
Good point, will do that in the next PR.